cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -82,9 +82,10 @@ public RemoteTaskExecutor(CConfiguration cConf, RemoteClientFactory remoteClient
   public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception {
     ByteBuffer requestBody = encodeTaskRequest(runnableTaskRequest);
 
-    return Retries.callWithRetries(() -> {
+    return Retries.callWithRetries((attempt) -> {
```

Comment:
It is strange to pass in the attempts to the remote. Is it just for emitting metrics purpose? Why not having the client to emit this metrics?

Reply:
Yes, it is just for emitting metrics. The advantage of emitting this in TaskWorkerHttpHandlerInternal is that all remote tasks metrics gets emitted regardless of individual clients. It is also simpler to analyze the success/failure case.

Reply:
They are different metrics, right? Client side knows how many times it retry (a failure can include not able to call the service). The task side should only emit metrics about the task itself.

Also, what do you mean by `The advantage of emitting this in TaskWorkerHttpHandlerInternal is that all remote tasks metrics gets emitted regardless of individual clients`?

Reply:
Makes sense that they are different metrics. Added client side metrics .  Attempts is not passed to the task worker any more. The original intention for adding metrics in Taskworker was to avoid having to add metrics in all individual clients, but since all requests go through RemoteTaskExecutor , added the client side metrics there.